### PR TITLE
fixed the ModuleDict key ordering problem

### LINKED ISSUE #40227: ModuleDict documentation misleadingly describes ordering of keys
## 📚 Documentation

The current documentation says it's an "ordered dictionary that respects the order of insertion", but it can, in fact, destroy its order. For example: 

image

We should update our documentation to clarify its behavior, especially as users might try to enumerate the dict sequentially. 

cc @jlin27 @albanD @mruberry

### PR DESCRIPTION
**Pull request by yl-to**
fix #40227
Remove the sorting operation both in ModuleDict class
Also remove a sort operation in corresponding unit test, which will lead to unit test fail.

BC Note: Python version after 3.6, the plain dict will preserve the order of keys.
example:
For a python 3.6+ user, if he is initial a ModuleDict instance using plain python dict:
{
"b": torch.nn.MaxPool2d(3),
"a": torch.nn.MaxPool2d(3)
}
, he will get a ModuleDict which preserve the order:
ModuleDict(
(b): MaxPool2d(kernel_size=3, stride=3, padding=0, dilation=1, ceil_mode=False)
(a): MaxPool2d(kernel_size=3, stride=3, padding=0, dilation=1, ceil_mode=False)
)

For a python 3.5 user, if we maintain the same input, then the output ModuleDict could be:
ModuleDict(
(a): MaxPool2d(kernel_size=3, stride=3, padding=0, dilation=1, ceil_mode=False)
(b): MaxPool2d(kernel_size=3, stride=3, padding=0, dilation=1, ceil_mode=False)
)

### CODE REVIEW

**Comment by albanD** [comment]
This is BC breaking right? Can you give an example of what will be changing for the user?

**Comment by colesbury** [comment]
Thanks for taking this on. The PR should also update the docs for ModuleDict:

https://pytorch.org/docs/master/generated/torch.nn.ModuleDict.html#torch.nn.ModuleDict

**Comment by yl-to** [comment]
> Thanks for taking this on. The PR should also update the docs for ModuleDict:
> 
> https://pytorch.org/docs/master/generated/torch.nn.ModuleDict.html#torch.nn.ModuleDict

Hi, I submitted new commit about documentation and python version problem mentioned there, please help to review, thanks!

**Comment by yl-to** [comment]
> This is BC breaking right? Can you give an example of what will be changing for the user?

Hi I think it is BC for future python version.
a example:
For a python 3.6+ user, if he is initial a ModuleDict instance using plain python dict: 
{
    "b": torch.nn.MaxPool2d(3),
    "a": torch.nn.MaxPool2d(3)
}
, he will get a ModuleDict which preserve the order:
ModuleDict(
  (b): MaxPool2d(kernel_size=3, stride=3, padding=0, dilation=1, ceil_mode=False)
  (a): MaxPool2d(kernel_size=3, stride=3, padding=0, dilation=1, ceil_mode=False)
)

For a python 3.5 user,  if we maintain the same input, then the output ModuleDict could be:
ModuleDict(
  (a): MaxPool2d(kernel_size=3, stride=3, padding=0, dilation=1, ceil_mode=False)
  (b): MaxPool2d(kernel_size=3, stride=3, padding=0, dilation=1, ceil_mode=False)
)

**Comment by albanD** [comment]
Sounds good! Can you add this in the PR description at the top please (a new paragraph about BC breaking changes)? This will be used when we do the release note to warn about BC breaking changes. Thanks!

**Comment by yl-to** [comment]
> Sounds good! Can you add this in the PR description at the top please (a new paragraph about BC breaking changes)? This will be used when we do the release note to warn about BC breaking changes. Thanks!

Updated, please let me know if this is what you are asking for. Thanks!

**Comment by albanD** [comment]
Looks good. Thanks!

**Review by albanD — APPROVED** [review]

**Comment by albanD** [comment]
Can you rebase on top of master to make sure the CI passes please?

**Comment by yl-to** [comment]
> Can you rebase on top of master to make sure the CI passes please?

Can you please provide a hint how to do the rebase?
Thanks!

**Comment by yl-to** [comment]
> Can you rebase on top of master to make sure the CI passes please?

Sorry am I doing the right process? I don't know if I added some unnecessary commits.

**Comment by yl-to** [comment]
Am creating a new PR for this
